Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -231,19 +338,21 @@ protected override void EndProcessing()
                     progressIds.Add(progressId);
                 }
 
-                if (ParameterSetName == "ParameterGroup" && !StartAfter.HasValue)
+                if (!StartAfter.HasValue)
                 {
-                    // Create BackgroundTask 
-                    WriteDebug($"Create BackgroundTask EventType[{EventType}] RawInputData.length[{RawInputData.Length}]");
+                    // Create BackgroundTask
+                    WriteDebug(ParameterSetName == "IshObjectsGroup" ?
+                        $"Create BackgroundTask EventType[{EventType}] InputData.length[{inputData.Length}]" :
+                        $"Create BackgroundTask EventType[{EventType}] RawInputData.length[{RawInputData.Length}]");
                     var newBackgroundTaskRequest = new CreateBackgroundTaskRequest
                     {
                         eventType = EventType,
-                        hashId = "",
-                        inputData = Encoding.Unicode.GetBytes(RawInputData),
+                        hashId = HashId,
+                        inputData = backgroundTaskInputData,
                         progressId = startEventResponse.progressId
                     };
-                    var createBackgroundTaskResponse = IshSession.BackgroundTask25.CreateBackgroundTask(newBackgroundTaskRequest);
-                    var progressId = createBackgroundTaskResponse.progressId;
+                    var createBackgroundTaskStartAfterResponse = IshSession.BackgroundTask25.CreateBackgroundTask(newBackgroundTaskRequest);
+                    var progressId = createBackgroundTaskStartAfterResponse.progressId;
                     progressIds.Add(progressId);
```

Comment:
I don't understand why we add this single value progressId to a list. Event with the DocumenObj25 call we are returning 1 taskId and 1 progressId. So can we remove this list and the code "if (progressIds.Count > 1)"
and for the following part an "FilterOperator.equal" and "valuetype.value" are slightly better: 
"new IshMetadataFilterField(FieldElements.BackgroundTaskProgressId,
                        Enumerations.Level.Task, Enumerations.FilterOperator.In, progressIds.First().ToString(),
                        Enumerations.ValueType.Element)"